lib/image-diff.js

```diff
@@ -8,25 +9,25 @@ var mkdirp = require('mkdirp');
 var tmp = require('tmp');
 
 // Define custom resize function
-// Taken from https://github.com/twolfson/twolfson.com/blob/3.4.0/test/perceptual-tests/twolfson.com_test.js#L88-L107
-// TODO: Make image resizing its own library
-// DEV: This does not pollute gm.prototype
-gm.prototype.fillFromTo = function (params) {
+function transparentExtent(gm, params) {
+  // Assert we received our parameters
+  assert.notEqual(params.width, undefined);
+  assert.notEqual(params.height, undefined);
```

Comment:
These assertions are for preventing typos from developers. There's not really a case where the error would come from user error:

- If the file doesn't exist, then we will get an `ENOENT` error during the earlier `fs.stat`/`fs.exists` call and we won't be upscaling the image
    - https://github.com/uber/image-diff/blob/1.3.0/lib/image-diff.js#L122-L128
    - https://github.com/uber/image-diff/blob/1.3.0/lib/image-diff.js#L132
    - https://github.com/uber/image-diff/blob/1.3.0/lib/image-diff.js#L33-L52
    - https://github.com/uber/image-diff/blob/1.3.0/lib/image-diff.js#L137
    - https://github.com/uber/image-diff/blob/1.3.0/lib/image-diff.js#L170-L173
- Otherwise, we will always be receiving `height`/`width` from `gm.size` so if those don't exist, then we have a deeper problem in `gm` itself